pio/lib/Sender/Sender.cpp

```diff
@@ -720,3 +720,175 @@ bool SenderClass::sendBrewblox(String server, uint16_t port, String topic, Strin
     stopclient();
     return response;
 }
+
+uint32_t SenderClass::sendBricks()
+{
+  uint32_t next_request_ms = 0;
+
+  // dump json
+  CONSOLELN(F("sendBricks called"));
+
+  serializeJson(_doc, Serial);
+  WiFiClientSecure client;
+
+  CONSOLELN(F("setting insecure"));
+  client.setInsecure(); // unfortunately necessary, ESP8266 does not support SSL without hard coding certificates
+  String url = "https://bricks.bierbot.com/api/iot/v1";
+  //String url = "http://192.168.2.108:5001/bierbot-cloud/us-central1/iot_v1";
+  client.connect(url, 443);
+  CONSOLELN(F("adding headers"));
+  HTTPClient http; //Declare an object of class HTTPClient
+
+  // configure traged server and uri
+  http.begin(client, url);
+  http.addHeader("User-Agent", "iSpindel");
+  http.addHeader("Connection", "close");
+  http.addHeader("Content-Type", "application/json");
+
+  String json;
+  serializeJson(_doc, json);
+  auto httpCode = http.POST(json);
+  CONSOLELN(String(F("code: ")) + httpCode);
+
+  // httpCode will be negative on error
+  if (httpCode > 0)
+  {
+    if (httpCode == HTTP_CODE_OK)
+    {
+        String payload = http.getString(); //Get the request response payload
+        CONSOLE(F("Received response: "));
+        CONSOLELN(payload);
+
+        // this whole section below is to spare the JSON parser
+        uint8 startIdx = payload.indexOf("next_request_ms") + 17;
+        String nrmSubstring = payload.substring(startIdx);
+        uint8 endIdx = -1;
+        if (nrmSubstring.indexOf("}") != -1)
+        {
+            // end based on bracket
+            endIdx = nrmSubstring.indexOf("}") + startIdx;
+        }
+        else if (nrmSubstring.indexOf(",") != -1)
+        {
+            // end based on comma, other value following
+            endIdx = nrmSubstring.indexOf(",") + startIdx;
+        }
+
+        if (startIdx > 0 && endIdx > startIdx)
+        {
+            CONSOLE(F("next request string in "));
+            String next_request_str = payload.substring(startIdx, endIdx);
+            CONSOLE(next_request_str);
+            CONSOLE(F("ms"));
+            next_request_ms = next_request_str.toInt();
+        }
+    }
+    }
+    else
+    {
+      CONSOLE(F("[HTTP] POST... failed, error: "));
+      CONSOLELN(http.errorToString(httpCode));
+    }
+
+    http.end();
+    stopclient();
+    return next_request_ms;
+
+    /*
+
+    // forge URL for GET request
```

Comment:
what is the purpose of all these commented lines, can we clean them up?

Reply:
yes, I can (and will do tonight).

Reply:
done